holmes.go

```diff
@@ -324,6 +361,76 @@ func (h *Holmes) cpuProfile(curCPUUsage int) bool {
 	return true
 }
 
+func (h *Holmes) gcHeapCheckAndDump() {
+	if !h.opts.GCHeapOpts.Enable {
+		return
+	}
+
+	memStats := new(runtime.MemStats)
+	runtime.ReadMemStats(memStats)
```

Comment:
ReadMemStats will trigger stw，if this is triggered in every gc cycle，will it cause a heavy cost on performance？🤔

Reply:
yeah, I noticed it the STW. I think we may introduce the `debug.GCHeapMarked` to avoid STW. I just added a comment for it.
Also, I think the STW is acceptable for now.

Reply:
some small memory app has high gc frequency, eg 5 or 10 times per second, triggering stw 10times in one second will affect the latency, 

> Our initial approach was to have a ticker to run every second to monitor the heap metrics, and then adjust GOGC value accordingly. The disadvantage of this approach is that the overhead starts to become considerable, because in order to read heap metrics Go needs to do a STW ([ReadMemStats](https://golang.org/pkg/runtime/#ReadMemStats)) and it is somewhat inaccurate, because we can have more than one garbage collection per second.

the above paragarph is quoted from https://eng.uber.com/how-we-saved-70k-cores-across-30-mission-critical-services/